advanced/sr_gui_fingertip_visualization/src/sr_gui_fingertip_visualization/sr_gui_fingertip_visualization.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import
+
+import rospy
+import rostopic
+import sys
+
+from python_qt_binding.QtCore import Qt, QTimer, QRectF
+from python_qt_binding.QtGui import QPainter, QColor
+from qt_gui.plugin import Plugin
+
+from python_qt_binding.QtWidgets import (
+    QWidget,
+    QApplication,
+    QTabWidget,
+    QVBoxLayout,
+    QBoxLayout,
+    QHBoxLayout,
+    QPushButton,
+    QComboBox,
+    QMessageBox,
+    QLabel
+)
+
+from sr_gui_fingertip_visualization.visual_tab_layouts import VisualizationTab
+from sr_gui_fingertip_visualization.graph_tab_layouts import GraphTab
+
+class SrFingertipVisualizer(Plugin):
+    TITLE = "Fingertip Visualizer"
+
+    def __init__(self, context):
+        super().__init__(context)
+        self._detect_hand_and_tactile_type()
+        self.context = context
+        self._init_ui()
+
+    def _detect_hand_and_tactile_type(self):
```

Comment:
A lot of questions about this function. But the main one is why not using the HandFinder as in the other plugins?